Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -53,24 +53,17 @@ public class ExecutionEngine : IDisposable
 
     public readonly VerificationResultCache Cache;
 
-    static readonly MemoryCache programCache = new MemoryCache("ProgramCache");
-
     static readonly CacheItemPolicy policy = new CacheItemPolicy
       { SlidingExpiration = new TimeSpan(0, 10, 0), Priority = CacheItemPriority.Default };
 
-    public static Program CachedProgram(string programId) {
-      var result = programCache.Get(programId) as Program;
-      return result;
-    }
-
     public ExecutionEngine(ExecutionEngineOptions options, VerificationResultCache cache) {
       Options = options;
       Cache = cache;
       checkerPool = new CheckerPool(options);
       verifyImplementationSemaphore = new SemaphoreSlim(Options.VcsCores);
       
       var largeThreadScheduler = CustomStackSizePoolTaskScheduler.Create(16 * 1024 * 1024, Options.VcsCores);
-      largeThreadTaskFactory = new(CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskContinuationOptions.None, largeThreadScheduler);
+      largeThreadTaskFactory = new(CancellationToken.None, TaskCreationOptions.None, TaskContinuationOptions.None, largeThreadScheduler);
```

Comment:
Do you know why it was `DenyChildAttach` before?

Reply:
I don't think this change has any effect. By default tasks are not attached. A child must specify `TaskCreationOptions.AttachedToParent` before it attaches, which isn't done in this codebase.